profile/profile_test.go

```diff
@@ -275,6 +275,53 @@ var cpuL = []*Location{
 	},
 }
 
+var cpuLFolded = []*Location{
```

Comment:
This seems to mostly duplicate `cpuL` profile from above setting IsFolded flag? Perhaps we could generate one profile from the other? For example, set IsFolded on some locations in `cpuL` in the test?

Or actually even produce testProfile1Folded from testProfile1 by massaging some of its locations to have the folded flag? This would save a lot of duplication unless the profiles you got are not actually exact copies of each other modulo the folded flag.

Reply:
The only tricky thing is setting the Location field in samples so samples reference the folded locations.

Reply:
The profile copying is deep, so I think it should work fine.

Reply:
Yep -- this works. Done!